nautobot/ipam/factory.py

```diff
@@ -111,3 +107,199 @@ def prefix(self, n):
+
+    # Name needs to be globally unique, but the random route-distinguisher generation space is large enough that
+    # we'll deal with collisions as and when they occur.
+    name = factory.LazyFunction(random_route_distinguisher)
+
+    has_description = factory.Faker("pybool")
+    description = factory.Maybe("has_description", factory.Faker("sentence"), "")
+
+    has_tenant = factory.Faker("pybool")
+    tenant = factory.Maybe("has_tenant", random_instance(Tenant), None)
+
+
+class VRFFactory(PrimaryModelFactory):
+    class Meta:
+        model = VRF
+        exclude = (
+            "has_description",
+            "has_rd",
+            "has_tenant",
+        )
+
+    # note that name is *not* globally unique
+    name = factory.Faker("color_name")
+
+    # RD needs to be globally unique, but the random route-distinguisher generation space is large enough that
+    # we'll deal with collisions as and when they occur.
+    has_rd = factory.Faker("pybool")
+    rd = factory.Maybe("has_rd", factory.LazyFunction(random_route_distinguisher), None)
+
+    has_tenant = factory.Faker("pybool")
+    tenant = factory.Maybe("has_tenant", random_instance(Tenant), None)
+
+    enforce_unique = factory.Faker("pybool")
+
+    has_description = factory.Faker("pybool")
+    description = factory.Maybe("has_description", factory.Faker("sentence"), "")
+
+    @factory.post_generation
+    def import_targets(self, create, extracted, **kwargs):
+        if create:
+            if extracted:
+                self.import_targets.set(extracted)
+            else:
+                self.import_targets.set(get_random_instances(RouteTarget))
+
+    @factory.post_generation
+    def export_targets(self, create, extracted, **kwargs):
+        if create:
+            if extracted:
+                self.export_targets.set(extracted)
+            else:
+                self.export_targets.set(get_random_instances(RouteTarget))
+
+
+class RoleFactory(OrganizationalModelFactory):
+    class Meta:
+        model = Role
+        exclude = ("has_description",)
+
+    name = factory.Faker("word", part_of_speech="adjective")
+
+    weight = factory.Faker("pyint", min_value=100, step=100)
+
+    has_description = factory.Faker("pybool")
+    description = factory.Maybe("has_description", factory.Faker("sentence"), "")
+
+
+class VLANGroupFactory(OrganizationalModelFactory):
+    class Meta:
+        model = VLANGroup
+        exclude = (
+            "has_description",
+            "has_location",
+            "has_site",
+        )
+
+    # TODO: name is not globally unique, but (site, name) tuple must be.
+    # The likelihood of collision with random names is pretty low, but non-zero.
+    # We might want to consider *intentionally* using non-globally-unique names for testing purposes?
+    name = factory.Faker("word", part_of_speech="noun")
+
+    has_description = factory.Faker("pybool")
+    description = factory.Maybe("has_description", factory.Faker("sentence"), "")
```

Comment:
Should we use `Faker("text")` instead of `sentence` so that we can set max characters to match the model's `max_length`?

Reply:
Good thought, will do.